practices/Ramon-OC/practica01/practica01.py

```diff
@@ -0,0 +1,332 @@
+          result += wordResult[0] + " "
+    return result
+
+
+# + [markdown] id="db2b7bfb"
+#
+#
+# 2. Agregar un modo de búsqueda donde dada una palabra te muestre sus *homofonos*[1]. Debe mostrar su representación IPA y la lista de homofonos (si existen)
+#
+# ```
+# [es_MX]>>> habares
+# /aβaɾes/
+# ['abares', 'habares', 'havares']
+# ```
+#
+# [1]: palabras con el mismo sonido pero distinta ortografía
+
+# + id="50a18dd8"
+def query_homophones(searchWord: str, dataset: dict) -> list[str]:
+    """Search for a word in an IPA phonetics dict
+
+      Given a sentence this function returns a string with its homophones
+
+      Parameters:
+      -----------
+      searchWord: str
+          A word to search in the dataset
+      dataset: dict
+          A dataset for a given language code
+      Returns
+      -------
+      str:
+          String with posible homophones.
+      """
+    ipa = query_ipa_transcriptions(searchWord, dataset);
+    homophones = []
+    if(ipa[0] != "NOT FOUND"):
+      for word in dataset:
+        if dataset[word] == ipa[0]:
+          homophones.append(word)
+
+    return homophones
+
+
+# + [markdown] id="4db5d5d4-39b5-4fb8-a582-546e0732e197"
+# #### EXTRA
+
+# + id="97125f31"
+def validate_word(word: str, dataset: dict) -> str:
+  """ Search for a word in an IPA phonetics dict, if it´s not present it shows multiple
+      suggestions
+
+      Parameters:
+      -----------
+      dataset: dict
+          A dataset for a given language code
+      Returns
+      -------
+      str:
+          String with a valid word for the dataset.
+      """
+  transcription = query_ipa_transcriptions(word, dataset)
+  if(transcription[0] != "NOT FOUND"):
+    return word
+  else:
+    while transcription[0] == "NOT FOUND":
+      sugerencias = get_close_matches(word, dataset.keys(), n=3, cutoff=0.6)
+      if sugerencias:
+        mensaje = f"No se encontró '{word}' en el dataset. Palabras aproximadas:"
+        for sugerencia in sugerencias:
+            mensaje += f"\n{sugerencia}"
+        print(mensaje)
+        word = input(f" word>> ")
+        transcription = query_ipa_transcriptions(word, dataset)
+      else:
+        print("No se encontraron palabras similares a '{word}' en el dataset.")
+        word = input(f" word>> ")
+        transcription = query_ipa_transcriptions(word, dataset)
+
+  return word
+
+
+# + [markdown] id="6e4de293"
+# ### Browser with multiple options:
+#
+# 1. Get IPA transcript (word)
+# 2. Get IPA transcript (phrase)
+# 3. Homophone search
+#
+
+# + colab={"base_uri": "https://localhost:8080/"} id="201bbfdb" outputId="eaa5a819-7647-40ab-9fb3-7f6255ca014f"
+def browser():
+    print("Lenguas disponibles:")
+    for lang_key in dataset.keys():
+      print(f"{lang_key}: {lang_codes[lang_key]}")
+
+    lang = input("lang>> ")
+    sub_data = dataset[lang]
+
+    print("1. Obtener transcripción IPA (palabra) \n2. Obtener transcripción IPA (frase)\n3. Búsqueda de homófonos")
```

Comment:
Es mejor que la línea no pase de 120 caracteres por línea
```suggestion
    print("1. Obtener transcripción IPA (palabra)"
    print("2. Obtener transcripción IPA (frase)")
    print("3. Búsqueda de homófonos")
```